tempodb/encoding/vparquet3/block_autocomplete.go

```diff
@@ -0,0 +1,869 @@
+			pred, err := createStringPredicate(cond.Op, cond.Operands)
+			if err != nil {
+				return nil, err
+			}
+			addPredicate(columnPathSpanID, pred)
+			columnSelectAs[columnPathSpanID] = columnPathSpanID
+			continue
+
+		case traceql.IntrinsicSpanStartTime:
+			pred, err := createIntPredicate(cond.Op, cond.Operands)
+			if err != nil {
+				return nil, err
+			}
+			addPredicate(columnPathSpanStartTime, pred)
+			columnSelectAs[columnPathSpanStartTime] = columnPathSpanStartTime
+			continue
+
+		case traceql.IntrinsicName:
+			pred, err := createStringPredicate(cond.Op, cond.Operands)
+			if err != nil {
+				return nil, err
+			}
+			addPredicate(columnPathSpanName, pred)
+			columnSelectAs[columnPathSpanName] = columnPathSpanName
+			continue
+
+		case traceql.IntrinsicKind:
+			pred, err := createIntPredicate(cond.Op, cond.Operands)
+			if err != nil {
+				return nil, err
+			}
+			addPredicate(columnPathSpanKind, pred)
+			columnSelectAs[columnPathSpanKind] = columnPathSpanKind
+			continue
+
+		case traceql.IntrinsicDuration:
+			pred, err := createIntPredicate(cond.Op, cond.Operands)
+			if err != nil {
+				return nil, err
+			}
+			addPredicate(columnPathSpanDuration, pred)
+			columnSelectAs[columnPathSpanDuration] = columnPathSpanDuration
+			continue
+
+		case traceql.IntrinsicStatus:
+			pred, err := createIntPredicate(cond.Op, cond.Operands)
+			if err != nil {
+				return nil, err
+			}
+			addPredicate(columnPathSpanStatusCode, pred)
+			columnSelectAs[columnPathSpanStatusCode] = columnPathSpanStatusCode
+			continue
+		case traceql.IntrinsicStatusMessage:
+			pred, err := createStringPredicate(cond.Op, cond.Operands)
+			if err != nil {
+				return nil, err
+			}
+			addPredicate(columnPathSpanStatusMessage, pred)
+			columnSelectAs[columnPathSpanStatusMessage] = columnPathSpanStatusMessage
+			continue
+
+		case traceql.IntrinsicStructuralDescendant:
+			selectColumnIfNotAlready(columnPathSpanNestedSetLeft)
+			selectColumnIfNotAlready(columnPathSpanNestedSetRight)
+
+		case traceql.IntrinsicStructuralChild:
+			selectColumnIfNotAlready(columnPathSpanNestedSetLeft)
+			selectColumnIfNotAlready(columnPathSpanParentID)
+
+		case traceql.IntrinsicStructuralSibling:
+			selectColumnIfNotAlready(columnPathSpanParentID)
+		}
+
+		// Well-known attribute?
+		if entry, ok := wellKnownColumnLookups[cond.Attribute.Name]; ok && entry.level != traceql.AttributeScopeResource {
+			if cond.Op == traceql.OpNone {
+				addPredicate(entry.columnPath, nil) // No filtering
+				columnSelectAs[entry.columnPath] = cond.Attribute.Name
+				continue
+			}
+
+			// Compatible type?
+			if entry.typ == operandType(cond.Operands) {
+				pred, err := createPredicate(cond.Op, cond.Operands)
+				if err != nil {
+					return nil, errors.Wrap(err, "creating predicate")
+				}
+				addPredicate(entry.columnPath, pred)
+				columnSelectAs[entry.columnPath] = cond.Attribute.Name
+				continue
+			}
+		}
+
+		// Attributes stored in dedicated columns
```

Comment:
There is a lot of overlap between this and the iterators for search.  Do you have any thoughts on if this can/should be addressed?   Could we expand the existing `createThingIterator` funcs to take the group predicates, which I believe are the main differences?

Reply:
The consensus is this isn't worth trying to deduplicate the code path for Autocomplete and Search.  It seems possible eventually but very complex and not worth tackling in this PR.